src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,192 @@
+    }
+    if (routeDescription.getEnd() != null && numTerritories > 1) {
+      points[numTerritories - 1] = new Point(routeDescription.getEnd());
+    }
+    return points;
   }
 
-  /**
-   * (x,y) - the first point to draw from (xx, yy) - the point to draw too
-   * (xxx, yyy) - the next point that the line segment will be drawn to
-   */
-  private static void drawCurvedLineWithNextPoint(final double x, final double y,
-      final double xx, final double yy, final double xxx, final double yyy, final List<Shape> shapes) {
-    final int maxControlLength = 150;
-    double controlDiffx = xx - xxx;
-    double controlDiffy = yy - yyy;
-    if (Math.abs(controlDiffx) > maxControlLength || Math.abs(controlDiffy) > maxControlLength) {
-      double ratio = 0.0;
-      try {
-        ratio = Math.abs(controlDiffx / controlDiffy);
-      } catch (final ArithmeticException ex) {
-        ratio = 1000;
-      }
-      if (Math.abs(controlDiffx) > Math.abs(controlDiffy)) {
-        controlDiffx = controlDiffx < 0 ? -maxControlLength : maxControlLength;
-        controlDiffy = controlDiffy < 0 ? (int) (-maxControlLength / ratio) : (int) (maxControlLength / ratio);
-      } else {
-        controlDiffy = controlDiffy < 0 ? -maxControlLength : maxControlLength;
-        controlDiffx = controlDiffx < 0 ? (int) (-maxControlLength * ratio) : (int) (maxControlLength * ratio);
-      }
+  protected static Tuple<double[], double[]> pointsToDoubleArrays(Point[] points) {
+    double[] xResult = new double[points.length];
+    double[] yResult = new double[points.length];
+    for (int i = 0; i < points.length; i++) {
+      xResult[i] = points[i].getX();
+      yResult[i] = points[i].getY();
+    }
+    return Tuple.of(xResult, yResult);
+  }
+
+  protected static double[] getCoords(PolynomialSplineFunction curve, double[] index) {
+    final double detailLevel = 1.0;//Tweak this value in order to achieve good rendering results
+    final double defaultCoordSize = index[index.length - 1];
+    final double[] coords = new double[(int)Math.round(detailLevel * defaultCoordSize) + 1];
+    final double stepSize = curve.getKnots()[curve.getKnots().length - 1] / coords.length;
+    double curValue = 0;
+    for (int i = 0; i < coords.length; i ++) {
+      coords[i] = curve.value(curValue);
+      curValue += stepSize;
+    }
+    return coords;
+  }
+
+  private static void drawMoveLength(Graphics2D graphics, RouteDescription routeDescription, Point[] points,
+      int xOffset, int yOffset, MapPanel view, boolean scrollWrapX, boolean scrollWrapY, int numTerritories,
+      String movementLeftForCurrentUnits) {
+    final double scale = view.getScale();
+    final Point cursorPos = points[points.length - 1];
+    final int xDir = cursorPos.x - points[numTerritories - 2].x;
+    final int textXOffset = xDir > 0 ? 6 : (xDir == 0 ? 0 : -14);
+
+    final int yDir = cursorPos.y - points[numTerritories - 2].y;
+    final int textYOffset = yDir > 0 ? 18 : -24;
```

Comment:
Please extract the numbers to named constants, on this line would be `18`, and `-24`. The goal would be to answer the question of what these number represent. There are quite a few of them in this function as well.